Source/Dafny/Plugin.cs

```diff
@@ -7,19 +7,30 @@
 namespace Microsoft.Dafny;
 
 /// <summary>
-/// This class is internal to Dafny. It wraps an assembly and an extracted configuration from this assembly,
+/// This class wraps an assembly and an extracted configuration from this assembly,
 /// The configuration provides the methods to parse command-line arguments and obtain Rewriters 
 /// </summary>
 public class Plugin {
-  public Configuration Configuration;
+  public PluginConfiguration PluginConfiguration;
 
-  private readonly string path;
+  private readonly string pluginPath;
   private readonly Assembly assembly;
   private System.Type[] rewriterTypes;
 
-  class AutomaticConfiguration : Configuration {
+  private Plugin(string path, string[] args) {
+    pluginPath = path;
+    assembly = Assembly.LoadFrom(pluginPath);
```

Comment:
Nitpick: I would even do this work in the `Load` method and then pass the assembly to the constructor.

Reply:
Great alternative. I like to put arguments to the class too. Note that it's `Assembly.LoadFrom`, not `Assembly.Load`.